chrome/content/zotero/containers/collectionTree.jsx

```diff
@@ -0,0 +1,2179 @@
+const { IntlProvider } = require('react-intl');
+const Tree = require('components/tree');
+const Icons = require('components/icons');
+const { getDragTargetOrient } = require('components/utils');
+const cx = require('classnames');
+const { Cc, Ci } = require('chrome');
+
+const noop = Promise.resolve;
+
+Zotero.CollectionTree = class CollectionTree extends React.Component {
+	static init(domEl, opts) {
+		Zotero.debug("Initializing React CollectionTree");
+		var ref;
+		let elem = (
+			<IntlProvider locale={Zotero.locale} messages={ZoteroPane.intlMessages}>
+				<CollectionTree ref={c => ref = c } {...opts} />
+			</IntlProvider>
+		);
+		ReactDom.render(elem, domEl);
+		ref.domEl = domEl;
+		
+		// We need to return ref before issuing select events
+		// which will trigger a ZoteroPane listener that requires
+		// ZoteroPane.collectionsView to be assigned
+		(async function() {
+			await ref.refresh();
+			var lastViewedID = Zotero.Prefs.get('lastViewedFolder');
+			if (lastViewedID) {
+				var selected = await ref.selectByID(lastViewedID);
+			}
+			if (!selected) {
+				await ref.selectByID('L' + Zotero.Libraries.userLibraryID);
+			}
+			ref.selectEventsSuppressed = false;
+			await ref.runListeners('load');
+			Zotero.debug("React CollectionTree initialized");
+		})();
+		
+		return ref;
+	}
+	
+	constructor(props) {
+		super(props);
+		this.itemTreeView = null;
+		this.itemToSelect = null;
+		this.hideSources = [];
+		
+		this._rows = [];
+		this._highlightedRows = {};
+		this._unregisterID = Zotero.Notifier.registerObserver(
+			this,
+			[
+				'collection',
+				'search',
+				'feed',
+				'share',
+				'group',
+				'trash',
+				'bucket'
+			],
+			'collectionTree',
+			25
+		);
+		this._containerState = {};
+		this._virtualCollectionLibraries = {};
+		this._trashNotEmpty = {};
+		this.refs;
+		this._selectEventsSuppressed = false;
+		this._focused = null;
+		
+		this.onLoad = this._createEventBinding('load', true, true);
+		this.onSelect = this._createEventBinding('select');
+		this.onRefresh = this._createEventBinding('refresh');
+	}
+	
+	componentDidUpdate() {
+		this.runListeners('refresh');
+	}
+
+	// Add a keypress listener for expand/collapse
+	handleKeyDown = (event) => {
+		// if (this.editingRow != -1) return; // In-line editing active
+		
+		var libraryID = this.getSelectedLibraryID();
+		if (!libraryID) return;
+		
+		if (event.key == '+' && !(event.ctrlKey || event.altKey || event.metaKey)) {
+			this.expandLibrary(libraryID, true);
+		}
+		else if (event.key == '-' && !(event.shiftKey || event.ctrlKey ||
+				event.altKey || event.metaKey)) {
+			this.collapseLibrary(libraryID);
+		}
+		else if ((event.key == 'Backspace' && Zotero.isMac) ||
+				event.key == 'Delete') {
+			var deleteItems = event.metaKey || (!Zotero.isMac && event.shiftKey);
+			ZoteroPane.deleteSelectedCollection(deleteItems);
+			event.preventDefault();
+			return;
+		}
+	}
+	
+	handleRowFocus = (treeRow) => {
+		this.focused = treeRow;
```

Comment:
I am reimplementing this part for the item tree due to a need for more complex keyboard navigation and multiple-selection with proposed naming changes. Since those changes are in tree.jsx level, they will need to be downported to collection tree too. I would rather do the renaming later, as part of those changes, unless you insist in this PR.